### PR TITLE
[webapi] Clean up unreferenced HTML files

### PR DESCRIPTION
**Pull request by kangxu**
Remove unreferenced files in suites including:
  tct-2dtransforms-css3-tests
  tct-3dtransforms-css3-tests
  tct-backgrounds-css3-tests
  tct-csp-w3c-tests
  tct-extra-html5-tests
  tct-fileapi-w3c-tests
  tct-fonts-css3-tests
  tct-forms-html5-tests
  tct-geodeny-w3c-tests
  tct-mediacapture-w3c-tests
  tct-navigationtiming-w3c-tests
  tct-sandbox-html5-tests
  tct-touchevent-w3c-tests
  tct-webgl-nonw3c-tests
  tct-workers-w3c-tests
  tct-xmlhttprequest-w3c-tests
  webapi-audiosystem-tizen-tests
  webapi-mediaserver-tizen-tests
  webapi-nativefilesystem-xwalk-tests
  webapi-shadowdom-w3c-tests
  webapi-webcl-nonw3c-tests

https://crosswalk-project.org/jira/browse/XWALK-4563

### CODE REVIEW

**Comment by zqzhang** [comment]
@chunywang seems there are many tests for XHR from W3C not in our testing. Could you please check why? For the XHR test suite, I plan to sync with latest W3C repo next. So we'd better check why we haven't used the W3C tests.

**Comment by chunywang** [comment]
OK

**Comment by chunywang** [comment]
OK.

**Comment by zqzhang** [comment]
Hmm, seems some of the files are referenced by others. With this patch applied, I used commands below to check:
```
for f in `git diff --name-only HEAD^ | awk -F \/ '{print $NF}'`; do
  echo . $f;
  grep $f -rl *;
done
```

File: webapi/tct-2dtransforms-css3-tests/2dtransforms/csswg/transform-blank-ref.html [hunk 1]
```diff
@@ -1,10 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-2dtransforms-css3-tests/2dtransforms/csswg/transform-blank-ref.html]: This reference file is referenced by

- webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/transform3d-scale-004-manual.html
- webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/transform3d-preserve3d-012-manual.html

It should be listed by tests*xml files as reftests.

**Comment by zqzhang** [comment]
I will send the rest via mail, @kangxu

File: webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/reference/transform-origin-ref.html [hunk 2]
```diff
@@ -1,27 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/reference/transform-origin-ref.html]: Referenced by
- webapi/tct-2dtransforms-css3-tests/tests.full.xml
- webapi/tct-2dtransforms-css3-tests/2dtransforms/csswg/transform-origin-manual.html

File: webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/reference/transform-scale-test-ref.html [hunk 3]
```diff
@@ -1,21 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/reference/transform-scale-test-ref.html]: Referenced by
- webapi/tct-2dtransforms-css3-tests/tests.xml
- webapi/tct-2dtransforms-css3-tests/tests.full.xml
- webapi/tct-2dtransforms-css3-tests/2dtransforms/csswg/transform-scale-test-manual.html

File: webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/reference/transforms-rotate-degree-90-ref.html [hunk 4]
```diff
@@ -1,22 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/reference/transforms-rotate-degree-90-ref.html]: Referenced by

- webapi/tct-2dtransforms-css3-tests/tests.full.xml
- webapi/tct-2dtransforms-css3-tests/2dtransforms/csswg/transforms-rotate-degree-90-manual.html

File: webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/reference/transforms-rotate-translate-scale-ref.html [hunk 5]
```diff
@@ -1,22 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-3dtransforms-css3-tests/3dtransforms/csswg/reference/transforms-rotate-translate-scale-ref.html]: Referenced by
- webapi/tct-2dtransforms-css3-tests/tests.full.xml
- webapi/tct-2dtransforms-css3-tests/2dtransforms/csswg/transforms-rotate-translate-scale-manual.html

File: webapi/tct-backgrounds-css3-tests/backgrounds/csswg/reference/background-size-009-ref.html [hunk 6]
```diff
@@ -1,15 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-backgrounds-css3-tests/backgrounds/csswg/reference/background-size-009-ref.html]: There is a same file at misc/webapi-uiautomation-xwalk-tests/css3-background/reference/background-size-009-ref.html, which referenced by
- misc/webapi-uiautomation-xwalk-tests/tests.xml
- misc/webapi-uiautomation-xwalk-tests/css3-background/csswg/background-size-009.html
- misc/webapi-uiautomation-xwalk-tests/css3-background/csswg/background-size-017.html
- misc/webapi-uiautomation-xwalk-tests/css3-background/csswg/background-size-014.html

But this file isn't referenced by tct-backgrounds-css3-tests

File: webapi/tct-csp-w3c-tests/csp/reference/csp_font-src_cross-origin_multi_allowed_two-ref.html [hunk 7]
```diff
@@ -1,18 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-csp-w3c-tests/csp/reference/csp_font-src_cross-origin_multi_allowed_two-ref.html]: ditto, referenced by
- misc/webapi-uiautomation-xwalk-tests/tests.xml

File: webapi/tct-csp-w3c-tests/csp/reference/csp_script-src_unsafe-inline_blocked_ext-ref.html [hunk 8]
```diff
@@ -1,9 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-csp-w3c-tests/csp/reference/csp_script-src_unsafe-inline_blocked_ext-ref.html]: ditto, referenced by
- misc/webapi-uiautomation-xwalk-tests/tests.xml

File: webapi/tct-forms-html5-tests/forms/w3c/submission/constraints/form-validation-validity-tooLong.html [hunk 9]
```diff
@@ -1,48 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/tct-forms-html5-tests/forms/w3c/submission/constraints/form-validation-validity-tooLong.html]: Referenced by
- webapi/tct-extra-html5-tests/tests.xml
- webapi/tct-extra-html5-tests/tests.full.xml

File: webapi/tct-xmlhttprequest-w3c-tests/xmlhttprequest/w3c/status-error.htm [hunk 10]
```diff
@@ -1,60 +0,0 @@
-<!doctype html>
```
Comment (zqzhang) [inline webapi/tct-xmlhttprequest-w3c-tests/xmlhttprequest/w3c/status-error.htm]: This file is referenced by

- webapi/tct-sse-w3c-tests/tests.xml
- webapi/tct-sse-w3c-tests/tests.full.xml
- webapi/tct-sse-w3c-tests/sse-py/w3c/README
- webapi/tct-sse-w3c-tests/sse/w3c/READM

However, there is no reference in tct-xmlhttprequest-w3c-tests; please check why.

File: webapi/webapi-nativefilesystem-xwalk-tests/nativefilesystem/Metadata_modificationTime.html [hunk 11]
```diff
@@ -1,68 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/webapi-nativefilesystem-xwalk-tests/nativefilesystem/Metadata_modificationTime.html]: There is also a webapi/tct-filesystemapi-w3c-tests/filesystemapi/Metadata_modificationTime.html

File: webapi/webapi-nativefilesystem-xwalk-tests/nativefilesystem/Metadata_size.html [hunk 12]
```diff
@@ -1,68 +0,0 @@
-<!DOCTYPE html>
```
Comment (zqzhang) [inline webapi/webapi-nativefilesystem-xwalk-tests/nativefilesystem/Metadata_size.html]: Same name as webapi/tct-filesystemapi-w3c-tests/filesystemapi/Metadata_modificationTime.html

**Comment by kangxu** [comment]
@zqzhang I use commands below to check, no referenced files are cleaned up:

#!/usr/bin/sh
for f in `git diff --name-only HEAD^`; do
  suite=`echo $f |awk -F \/ '{print $2}'`
  x=`echo $f |awk -F \/ '{print $NF}'`
  cd $suite
  count=`grep $x -rl * |wc -l`
  if [ $count -ne 0 ];then
    echo $f
  fi
  cd - &>/dev/null
done

And I have also checked the result of  your commands, no referenced files are deleted also.

File: webapi/webapi-nativefilesystem-xwalk-tests/nativefilesystem/Metadata_size.html (hunk 12 above)
Comment (zqzhang) [inline webapi/webapi-nativefilesystem-xwalk-tests/nativefilesystem/Metadata_size.html]: Same file named as webapi/tct-filesystemapi-w3c-tests/filesystemapi/Metadata_size.html

**Comment by kangxu** [comment]
@zqzhang , no referenced files are cleaned up, these suspicious HTML files are not referenced by files in the same suite.

**Comment by zqzhang** [comment]
> I have checked these test cases, some of them even lack of script files, for example
> 
> xmlhttprequest-timeout-worker-overridesexpires.html lacks of 'resources/xmlhttprequest-timeout-overridesexpires.js'
> xmlhttprequest-timeout-worker-twice.html lack of 'resources/xmlhttprequest-timeout-twice.js'

@chunywang these 2 files are in this repo:

- webapi/tct-xmlhttprequest-w3c-tests/xmlhttprequest/w3c/xmlhttprequest-timeout-overridesexpires.js
- webapi/tct-xmlhttprequest-w3c-tests/xmlhttprequest/w3c/xmlhttprequest-timeout-twice.js

But they are NOT in `resources/`. Is this the problem why these tests are not used?

**Comment by chunywang** [comment]
No, i mean these case integrate file incomplete.

The tct-xmlhttprequest-w3c-tests suit has two  folders: xmlhttprequest and xmlhttpreques-py, these  cases are not contained in xmlhttpreques-py folder. It synchronize the xmlhttpreques-py folder's cases, so there are not tests*.xml.

**Comment by zqzhang** [comment]
> No, i mean these case integrate files incomplete.

OK, this is another issue. I leave that to you when you are ready to check W3C's XHR test suite and to integrate them back then.

**Pull request merged**